plugins/woocommerce/tests/e2e-pw/tests/merchant/create-variable-product.spec.js

```diff
@@ -55,19 +55,28 @@ test.describe.serial( 'Add New Variable Product Page', () => {
 			if ( i > 0 ) {
 				await page.click( 'button.add_attribute' );
 			}
-			const input = `input[name="attribute_names[${ i }]"]`;
 
-			await page.waitForSelector( input, { timeout: 1000 } ); // Wait for up to 1 second
-			await page.fill( input, `attr #${ i + 1 }` );
-			await page.fill(
-				`textarea[name="attribute_values[${ i }]"]`,
-				'val1 | val2'
-			);
+			await page
+				.locator(
+					`.woocommerce_attribute_data input[name="attribute_names[${ i }]"]`
```

Comment:
The selector `input[name="attribute_names[${ i }]"]` matches more than one element in the DOM (there is a hidden input with the same name). So, I added the `.woocommerce_attribute_data` class to make it more specific.